predicate/solver/test/test_teleport_access_requests.py

```diff
@@ -57,54 +68,276 @@ def test_access_requests(self):
         )
         assert ret is True, "Devs can review other folks access to stage"
 
-        # Can user with these policies review a role?
+        # Can user with these policies review a policy?
         ret, _ = devs.query(
             Review(
                 RequestPolicy.names.contains(
                     "access-prod",
                 )
             )
         )
-        assert ret is False, "can't review role that is not listed in the policy"
+        assert ret is False, "can't review policy that is not listed in the policy"
 
-        # TODO: how to bind to roles?
         ret, _ = devs.query(
             Request(
                 (RequestPolicy.names == ("access-stage",))
-                & (RequestPolicy.approvals["access-stage"] == ("alice", "bob"))
+                # Notice we are using full approval definition here using == to other map
+                & (
+                    RequestPolicy.approvals
+                    == StringSetMap("reviews", {"access-stage": ("alice", "bob")})
+                )
             )
         )
         assert ret is True, "two folks have approved the request"
 
         ret, _ = devs.query(
             Request(
                 (RequestPolicy.names == ("access-stage",))
-                & (RequestPolicy.approvals["access-stage"] == ("alice", "bob"))
-                & (RequestPolicy.denials["access-stage"] == ("ketanji",))
+                # Notice we are using full approval definition here using == to other map
+                & (
+                    RequestPolicy.approvals
+                    == StringSetMap("reviews", {"access-stage": ("alice",)})
+                )
+            )
+        )
+        assert ret is False, "only one person has approved the request"
+
+        # TODO: it's easy to make a mistake and have the same name for StringSetMap
```

Comment:
Would creating an issue to track this and referencing the ID be warranted here?

Reply:
done.